django/db/migrations/operations/models.py

```diff
@@ -534,67 +532,37 @@ def references_field(self, model_name, name, app_label=None):
+        return "Alter %s for %s (%s constraint(s))" % (
+            self.option_name, self.name, len(getattr(self, self.option_name) or [])
+        )
 
 
-class AlterIndexTogether(FieldRelatedOptionOperation):
+class AlterUniqueTogether(AlterFooTogether):
     """
-    Changes the value of index_together to the target one.
-    Input value of index_together must be a set of tuples.
+    Changes the value of unique_together to the target one.
+    Input value of unique_together must be a set of tuples.
     """
-    option_name = "index_together"
+    option_name = 'unique_together'
 
-    def __init__(self, name, index_together):
-        index_together = normalize_together(index_together)
-        self.index_together = set(tuple(cons) for cons in index_together)
-        super(AlterIndexTogether, self).__init__(name)
-
-    def deconstruct(self):
-        kwargs = {
-            'name': self.name,
-            'index_together': self.index_together,
-        }
-        return (
-            self.__class__.__name__,
-            [],
-            kwargs
-        )
-
-    def state_forwards(self, app_label, state):
-        model_state = state.models[app_label, self.name_lower]
-        model_state.options[self.option_name] = self.index_together
-        state.reload_model(app_label, self.name_lower)
-
-    def database_forwards(self, app_label, schema_editor, from_state, to_state):
-        new_model = to_state.apps.get_model(app_label, self.name)
-        if self.allow_migrate_model(schema_editor.connection.alias, new_model):
-            old_model = from_state.apps.get_model(app_label, self.name)
-            schema_editor.alter_index_together(
-                new_model,
-                getattr(old_model._meta, self.option_name, set()),
-                getattr(new_model._meta, self.option_name, set()),
-            )
+    def __init__(self, name, unique_together):
+        super(AlterUniqueTogether, self).__init__(name, unique_together)
 
-    def database_backwards(self, app_label, schema_editor, from_state, to_state):
-        return self.database_forwards(app_label, schema_editor, from_state, to_state)
 
-    def references_field(self, model_name, name, app_label=None):
-        return (
-            self.references_model(model_name, app_label) and
-            (
-                not self.index_together or
-                any((name in together) for together in self.index_together)
-            )
-        )
+class AlterIndexTogether(AlterFooTogether):
+    """
+    Changes the value of index_together to the target one.
+    Input value of index_together must be a set of tuples.
+    """
+    option_name = 'index_together'
 
-    def describe(self):
-        return "Alter %s for %s (%s constraint(s))" % (self.option_name, self.name, len(self.index_together or ''))
+    def __init__(self, name, index_together):
+        super(AlterIndexTogether, self).__init__(name, index_together)
```

Comment:
The signatures are different. If the value of `index_together` is passed as a keyword argument then it won't work.
For example, although `AlterIndexTogether('foo', ('a', 'b'))` would work fine, something like the following will throw an error
```python
AlterIndexTogether(
    name='foo',
    index_together=('a', 'b'),
)
```
because the signature of `AlterFooTogether` contains `value` instead of `index_together`.